internal/backend/db/dbgorm/values.go

```diff
@@ -0,0 +1,90 @@
+package dbgorm
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/protobuf/proto"
+	"gorm.io/gorm"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/db/dbgorm/scheme"
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+const maxValueSize = 64 * 1024
+
+func (db *gormdb) SetValue(ctx context.Context, envID sdktypes.EnvID, key string, v sdktypes.Value) error {
+	if !envID.IsValid() {
+		return sdkerrors.NewInvalidArgumentError("invalid env_id")
+	}
+
+	if v.ProtoSize() > maxValueSize {
+		return sdkerrors.NewInvalidArgumentError("value too large > %d bytes", maxValueSize)
+	}
+
+	uid, err := userIDFromContext(ctx)
```

Comment:
@itayd
please add tests using the convension you could see in other dbgorm-related files, e.g. extract setValue which should work with schema and add tests to it.

Reply:
done black box testing.